### PR TITLE
Add docs for FFT diffraction

### PR DESCRIPTION
**Pull request by g-braeunlich**

### CODE REVIEW

File: doc/validation/diffraction.rst [hunk 1]
```diff
@@ -252,6 +255,124 @@ and :math:`\boldsymbol{e}_N = \boldsymbol{e}_r(0) \times \boldsymbol{e}_E`.
 
 .. [1] https://calgary.rasc.ca/field_rotation.htm
 
+
+FFT Mode
+^^^^^^^^
+
+Conceptionally, to model diffraction in FFT mode, the image is
+convoluted with the PSF of the diffraction effect. This is implemented
```
Comment (cwwalter) [inline doc/validation/diffraction.rst:263]: convolved

File: doc/validation/diffraction.rst [hunk 2]
```diff
@@ -252,6 +255,124 @@ and :math:`\boldsymbol{e}_N = \boldsymbol{e}_r(0) \times \boldsymbol{e}_E`.
...
+by rendering the PSF to a finite square and convolve the square which each
```
Comment (cwwalter) [inline doc/validation/diffraction.rst:264]: for a single pixel and then convolving that pattern with a subset of the pixels in the image.

File: doc/validation/diffraction.rst [hunk 3]
```diff
@@ -252,6 +255,124 @@ and :math:`\boldsymbol{e}_N = \boldsymbol{e}_r(0) \times \boldsymbol{e}_E`.
...
+
+   \rho(r) = \frac{2}{R_0\pi}\frac{1}{1 + (r / R_0)^2}.
+
+The PSF for zero exposure time (no field rotation) is then defined as:
+
+.. math::
+
+   PSF_{\alpha=0}(x, y) = \frac{1}{4} \rho(\sqrt{x^2+y^2}) (\delta(x) + \delta(y)).
+
+Before rendered and convolved, a rotation corresponding to
```
Comment (cwwalter) [inline doc/validation/diffraction.rst:297]: Before being rendered and convolved,

**Review by cwwalter — APPROVED** [review]
I left some English suggestions and also a couple of questions where things were unclear to me. 

It basically looks OK for now  to me if we make those changes.

File: doc/validation/diffraction.rst (hunk 2 above)
Comment (g-braeunlich) [reply doc/validation/diffraction.rst]: Changed to 

> This is implemented by convolving the pattern for a single pixel with a subset of
> the pixels in the image.
> For each object a region of saturated pixels ...

But is not now the 2nd sentence a bit redundant?

File: doc/validation/diffraction.rst [hunk 4]
```diff
@@ -252,6 +255,124 @@ and :math:`\boldsymbol{e}_N = \boldsymbol{e}_r(0) \times \boldsymbol{e}_E`.
...
+
+.. math::
+
+   \rho(r) = \frac{2}{R_0\pi}\frac{1}{1 + (r / R_0)^2}.
+
+The PSF for zero exposure time (no field rotation) is then defined as:
+
+.. math::
+
+   PSF_{\alpha=0}(x, y) = \frac{1}{4} \rho(\sqrt{x^2+y^2}) (\delta(x) + \delta(y)).
```
[the remark this answers is not in the document]
Comment (g-braeunlich) [reply doc/validation/diffraction.rst:296]: This is a mathematical model for a thin, decaying cross. Important here is that
`\int\int PSF_{\alpha=0}(x, y) dx dy = 1`.
Do you have something different in mind, without using delta functions?

File: doc/validation/diffraction.rst [hunk 5]
```diff
@@ -252,6 +255,124 @@ and :math:`\boldsymbol{e}_N = \boldsymbol{e}_r(0) \times \boldsymbol{e}_E`.
...
+
+Before rendered and convolved, a rotation corresponding to
+``rotTelPos`` is applied.
+
+Diffraction PSF for non-zero exposure time
+``````````````````````````````````````````
+
+We neglect the time dependence of the rotation rate of the field
```
[the remark this answers is not in the document]
Comment (g-braeunlich) [reply doc/validation/diffraction.rst]: Changed but changed "we apply a coordinate transformation as follows" to a more detailed explanation, that the field rotation PSF is an integral over rotated copies of PSF_0

File: doc/validation/diffraction.rst [hunk 6]
```diff
@@ -252,6 +255,124 @@ and :math:`\boldsymbol{e}_N = \boldsymbol{e}_r(0) \times \boldsymbol{e}_E`.
...
+rotation and model the PSF according to
```
[the remark this answers is not in the document]
Comment (g-braeunlich) [reply doc/validation/diffraction.rst]: \alpha is the total field rotation angle. As we assume a constant rotation rate, \alpha is proportional to the exposure time.
This should now be explained.

File: doc/validation/diffraction.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (jmeyers314) [reply doc/validation/diffraction.rst:296]: I think the deltas are good.  You may want to add a note to the text that this makes a cross.

**Review by jmeyers314 — APPROVED** [review]
LGTM.  Note, I pushed a couple minor wording changes.

File: doc/validation/diffraction.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (g-braeunlich) [reply doc/validation/diffraction.rst:300]: OK. Done

**Comment by cwwalter** [comment]
@g-braeunlich Go ahead and merge this if you think it has now addressed the comments from Josh.  

I need to merge it into my changes.

**Pull request merged**